File: app/src/main/java/com/frankly/randomdog/MainActivity.java
package com.frankly.randomdog;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

/**
 * @author: ArshVardhanAtreya <arshvardhan@yahoo.co.in>
 */

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void generateDog(View view) {
        Intent dogGeneratorIntent = new Intent(this, DogGeneratorActivity.class);
        startActivity(dogGeneratorIntent);
    }

    public void recentlyGenerateDog(View view) {
        Intent dogGeneratorIntent = new Intent(this, RecentlyGeneratedDogActivity.class);
        startActivity(dogGeneratorIntent);
    }
}

File: app/src/main/java/com/frankly/randomdog/MainActivity.kt
package com.frankly.randomdog

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.frankly.randomdog.ui.doggenerator.view.DogGeneratorActivity
import com.frankly.randomdog.ui.recentlygenerateddog.view.RecentlyGeneratedDogActivity

/**
 * @author: ArshVardhanAtreya <arshvardhan></arshvardhan>@yahoo.co.in>
 */
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun generateDog(view: View?) {
        val dogGeneratorIntent = Intent(this, DogGeneratorActivity::class.java)
        startActivity(dogGeneratorIntent)
    }

    fun recentlyGenerateDog(view: View?) {
        val dogGeneratorIntent = Intent(this, RecentlyGeneratedDogActivity::class.java)
        startActivity(dogGeneratorIntent)
    }
}